generatebundlefile/hack/update_bundles.sh

```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+BASE_DIRECTORY=$(git rev-parse --show-toplevel)
+# Function to display usage information
+function usage() {
+  echo "Usage: $0 -e <environment>"
+  echo "  -e    Specify the environment (staging or prod)"
+  exit 1
+}
+
+# Function to check if required tools are installed
+function check_requirements() {
+  local required_tools=("aws" "jq" "yq")
+  for tool in "${required_tools[@]}"; do
+    if ! command -v "$tool" &>/dev/null; then
+      echo "Error: $tool is not installed or not in PATH" >&2
+      exit 1
+    fi
+  done
+}
+
+# Function to get the latest version from ECR
+function get_latest_version() {
+  local repo=$1
+  local k8s_version=$2
+
+  local json_output=$(aws ecr describe-images \
+    --repository-name "$repo" \
+    --output json)
+
+  # For cluster autoscaler filter tag based on k8s version (1.28,1.29,etc)
+  if [ "$repo" = "cluster-autoscaler/charts/cluster-autoscaler" ]; then
+    output=$(echo $json_output | jq -r --arg kv "1.$k8s_version" '
+    .imageDetails
+    | map(select(.imageTags | length > 0))
+    | map(select(.imageTags | map(test($kv)) | any))
+    | sort_by(.imagePushedAt)
+    | last
+    | .imageTags')
+  # For metrics-server filter tag based on k8s version (1-28,1-29,etc)
+  elif [ "$repo" = "metrics-server/charts/metrics-server" ]; then
+    output=$(echo $json_output | jq -r --arg kv "1-$k8s_version" '
+    .imageDetails
+    | map(select(.imageTags | length > 0))
+    | map(select(.imageTags | map(test($kv)) | any))
+    | sort_by(.imagePushedAt)
+    | last
+    | .imageTags')
```

Comment:
We can optimize this to not have this repeated code right?

Reply:
Updated.